Common/src/containers/CLookUpTable.cpp

```diff
@@ -117,41 +193,84 @@ void CLookUpTable::PrintTableInfo() {
+    for(auto i_level=0ul; i_level<n_table_levels; i_level++){
+      n_points_av += n_points[i_level] / n_table_levels;
+      n_tria_av += n_triangles[i_level] / n_table_levels;
+      n_edges_av += edges[i_level].size() / n_table_levels;
+      min_x = min(min_x, limits_table_x[i_level].first);
+      min_y = min(min_y, limits_table_y[i_level].first);
+      max_x = max(max_x, limits_table_x[i_level].second);
+      max_y = max(max_y, limits_table_y[i_level].second);
+    }
+    switch (table_dim)
+    {
+    case 2:
+      cout << "| Number of points:" << setw(47) << right << n_points_av << " |" << endl;
+      cout << "| Number of triangles:" << setw(44) << right << n_tria_av << " |" << endl;
+      cout << "| Number of edges:" << setw(48) << right << n_edges_av << " |" << endl;
+      cout << "+------------------------------------------------------------------+" << endl;
+      cout << "| Minimum Y:" << setw(54) << right << min_y << " |" << endl;
+      cout << "| Maximum Y:" << setw(54) << right << max_y << " |" << endl;
+      cout << "| Minimum X:" << setw(54) << right << min_x << " |" << endl;
+      cout << "| Maximum X:" << setw(54) << right << max_x << " |" << endl;
+      cout << "+------------------------------------------------------------------+" << endl;
+      cout << "| Variable names:                                                  |" << endl;
+      cout << "|                                                                  |" << endl;
+
+      for (unsigned long i_var = 0; i_var < names_var.size(); i_var++)
+        cout << "| " << right << setw(3) << i_var << ": " << left << setw(59) << names_var[i_var] << " |" << endl;
+
+      cout << "+------------------------------------------------------------------+" << endl;
+
+      cout << endl;
+      break;
+    case 3:
+      
+      cout << "| Number of table levels:" << setw(41) << right << n_table_levels << " |" << endl;
+      cout << "| Average number of points:" << setw(39) << right << floor(n_points_av) << " |" << endl;
+      cout << "| Average number of triangles:" << setw(36) << right << floor(n_tria_av) << " |" << endl;
+      cout << "| Average number of edges:" << setw(40) << right << floor(n_edges_av) << " |" << endl;
+      cout << "+------------------------------------------------------------------+" << endl;
+      cout << "| Minimum Z:" << setw(54) << right << limits_table_z.first << " |" << endl;
+      cout << "| Maximum Z:" << setw(54) << right << limits_table_z.second << " |" << endl;
+      cout << "| Minimum Y:" << setw(54) << right << min_y << " |" << endl;
+      cout << "| Maximum Y:" << setw(54) << right << max_y << " |" << endl;
+      cout << "| Minimum X:" << setw(54) << right << min_x << " |" << endl;
+      cout << "| Maximum X:" << setw(54) << right << max_x << " |" << endl;
+      cout << "+------------------------------------------------------------------+" << endl;
```

Comment:
Would it be too verbose if you printed the information per level? It would re-use the code for 1 level and maybe be more helpful for debugging.

Reply:
For small tables, maybe. I'm using the CLookUpTable class with 100+ levels. In such cases it would get too verbose in my opinion. I could make it a conditional statement where if the number of levels is like below 10, it will display all level info.

Reply:
I think it's ok to do just statistics then.